web/src/App.jsx

```diff
@@ -0,0 +1,231 @@
+import React from "react";
+import { create } from "zustand";
+import { init as initApm } from "@elastic/apm-rum";
+// import { useCookies } from "react-cookie";
+import "@/App.scss";
+import Home from "@/pages/Home/Home";
+import Player from "@/pages/Player/Player";
+import MenuBar from "./components/MenuBar/MenuBar";
+import produce from "immer";
+import { applyPlayerVol } from "./components/CardVolumeSlider/CardVolumeSlider";
+import { persist, createJSONStorage } from "zustand/middleware";
+import Settings from "@/pages/Settings/Settings";
+import { getSourceZones } from "@/pages/Home/Home";
+import DisconnectedIcon from "./components/DisconnectedIcon/DisconnectedIcon";
+
+import PropTypes from "prop-types";
+
+// const UPDATE_INTERVAL = 1000; Commented out while unused
+
+try {
+    const r = await fetch("/debug");
+    const debugDocText = await r.text();
+    const debugDoc = JSON.parse(debugDocText);
+    const debug = (debugDoc && debugDoc["debug"]) ? true : false;
+
+    var apm = initApm({ // eslint-disable-line no-unused-vars
+        active: debug,
+        serviceName: "Amplipi",
+        serverUrl: debugDoc["apmHost"] ? debugDoc["apmHost"] : "",
```

Comment:
@jonahshader I use top level await here to instantiate a very early connection to an arbitrary APM server, specified from a debug endpoint. I could also resolve the promise myself, or use a fancy vite plugin to transform top level awaits into resolved ones. I just didn't see a reason not to track more modern browsers and it seemed least messy at the time.